app/bundles/EmailBundle/Model/SendEmailToContact.php

```diff
@@ -292,6 +292,11 @@ public function reset()
         $this->mailer->reset();
     }
 
+    public function setSampleMailer()
```

Comment:
This will cause all subsequent uses of this class to bypass the default mailer. The original mailer should be stored then restored after the class's use. 

Also please add tests around these new methods.

Reply:
Work in progress.

Reply:
Try  current  solution:
- added $this->temporaryMailer 
- and re-save it before mailHelper reset

Do you have documentation to write tests for Mautic?

Reply:
It's not that different than writing PHPUNIT tests for any other PHP app. This model already has some tests:

https://github.com/mautic/mautic/blob/409eb114c3057b71022de5a76cebb43066f4931e/app/bundles/EmailBundle/Tests/Model/SendEmailToContactTest.php

Just make sure your changes are covered. Here is the command to run the tests: https://github.com/mautic/mautic/tree/409eb114c3057b71022de5a76cebb43066f4931e#phpunit